### PR TITLE
Adds download_jobs method and cli interface

### PR DESCRIPTION
**Pull request by nicolastomeo**

### CODE REVIEW

File: canotic/apis/jobs.py [hunk 1]
```diff
@@ -104,6 +104,33 @@ def list_jobs(self, app_id: str, page: int = None, size: int = None, sortBy: str
...
+        if completedStartDate is not None:
+            query_params['completedStartDate'] = completedStartDate.strftime('%Y-%m-%dT%H:%M:%SZ')
+        if completedEndDate is not None:
+            query_params['completedEndDate'] = completedEndDate.strftime('%Y-%m-%dT%H:%M:%SZ')
+        if statusIn is not None:
+            query_params['statusIn'] = statusIn
+        return self.request(uri, method='POST', query_params=query_params, required_api_key=True)
+
     def get_all_jobs(self, app_id: str, sortBy: str = 'id', orderBy: str = 'asc', createdStartDate: datetime = None,
                      createdEndDate: datetime = None,
```
Comment (ralfbanisch) [inline canotic/apis/jobs.py]: Just looking at this, customers are probably wondering what the difference between 'download_jobs' and 'get_all_jobs' is. Should we add a comment saying that 'get_all_jobs' does not include the job responses? Are there other significant differences?

File: canotic/apis/jobs.py [hunk 2]
```diff
@@ -104,6 +104,33 @@ def list_jobs(self, app_id: str, page: int = None, size: int = None, sortBy: str
             query_params['statusIn'] = statusIn
         return self.request(uri, method='GET', query_params=query_params, required_api_key=True)
 
+    def download_jobs(self, app_id: str, createdStartDate: datetime = None, createdEndDate: datetime = None,
+                      completedStartDate: datetime = None, completedEndDate: datetime = None,
+                      statusIn: List[str] = None) -> dict:
```
Comment (ralfbanisch) [inline canotic/apis/jobs.py]: I thought this triggers the email. Does it actually return a dict with the response or does it return nothing and the customer receives the email and clicks on the download link, as before?

File: canotic/apis/jobs.py (hunk 1 above)
Comment (nicolastomeo) [reply canotic/apis/jobs.py]: No other differences. i'll add that

File: canotic/apis/jobs.py (hunk 2 above)
Comment (nicolastomeo) [reply canotic/apis/jobs.py]: RIght it actually just return a string saying 'Task is processing' i'll fix the typing and docs

**Pull request merged**